src/query/server/server.go

```diff
@@ -727,25 +736,40 @@ func remoteClient(
 }
 
 func startGrpcServer(
-	logger *zap.Logger,
 	storage m3.Storage,
+	queryContextOptions models.QueryContextOptions,
 	poolWrapper *pools.PoolWrapper,
 	cfg *config.RPCConfiguration,
+	logger *zap.Logger,
 ) (*grpc.Server, error) {
 	logger.Info("creating gRPC server")
-	server := tsdbRemote.CreateNewGrpcServer(storage, poolWrapper)
-	waitForStart := make(chan struct{})
-	var startErr error
+	server := tsdbRemote.CreateNewGrpcServer(storage,
+		queryContextOptions, poolWrapper)
+
+	var (
+		waitForStart = make(chan struct{}, 1)
```

Comment:
Can you just make this a `chan error` and get rid of the startErr atomic stuff? Seems confusing for no reason.

```golang
// StartNewGrpcServer starts server on given address, then notifies channel
func StartNewGrpcServer(
	server *grpc.Server,
	address string,
	waitForStart chan<- error{},
) error {
	lis, err := net.Listen("tcp", address)
	if err != nil {
                 waitForStart <-err
	}

	waitForStart <- nil
	return server.Serve(lis)
}
```

Reply:
The whole thing is kinda messed up, I wanted to introduce least amount of changes possible.

I'll just break it up into two steps and make the grpc server take the listener.